Adding Kibana dashboard

paulRbr opened the pull request:
:warning: Work In Progress :warning: 

**Hello!**

You did a great job with this cartridge, thanks! I am adding a kibana dashboard in there, what do you think?

Check it live here: http://es-pbonaud.rhcloud.com/index.html

paulRbr committed:
Update manifest and add a port to serve kibana

paulRbr committed:
Add kibana installation in the cartridge

paulRbr committed:
Update README.md

caruccio commented:
Thanks A LOT for this PR. We do really appreciate your contribution!

There seams to be a missing configuration somewhere. After creating the app, it runs ES smooth, however Kibana is not loading it's frontend resources from /kibana, but / only - 404 everywhere 

![captura de tela 2014-11-13 as 20 11 49](https://cloud.githubusercontent.com/assets/540936/5037818/919b2518-6b71-11e4-81bd-f680d2468d43.png)

The funny part is that `index.html` has relative URLs only, thus `img/small.png` should become `/kibana/img/small.png`, AFAIK.

Do you have any idea what is going on here os is it just me?

paulRbr committed:
Add nginx as a real webserver

paulRbr committed:
Serve kibana directory too

paulRbr committed:
Adapt README to be consistent with PR for getupcloud

paulRbr commented:
Hi @caruccio,

Happy to hear that it's something you'd like to see :)

I changed the webserver from when you tested, and switched to nginx (inspired by https://github.com/gsterjov/openshift-nginx-cartridge).

**So now** ElasticSearch is not exposed to the public, but there is an nginx web server where by default, requests are proxied to ElasticSearch. This enables me to serve the kibana files and for users to be able to add server configuration such as basic auth, ip whitelisting... or more.

You can test directly on the master of my fork https://github.com/swcc/openshift-cartridge-elasticsearch

Let me know if you see some bad practices / errors in here.

caruccio commented on template/nginx.conf.erb:
```diff
@@ -0,0 +1,14 @@
+        proxy_set_header Host      $host;
+        proxy_set_header X-Real-IP $remote_addr;
+    }
+    location / {
```
Hi @popox.
Now **you** are doing a great job :+1: 

I've found just a little issue in nginx config. For some obscure reason, it is serving ES from URL `/`, ignoring the `index` directive. In order to reach kibana, one must request `/index.html`, which seams a little cumbersome.

I could bypass it by adding this before `location /`:

``` nginx
    # Kibana URI is /kibana
    location ~ ^/kibana/?$ {
        try_files /index.html =404;
    }
```

What do you think?

paulRbr replied:
Yes I thought about it, I was not sure what would be the best practice.

I guess serving kibana on the `/kibana` route would make much more sense than have to type `/index.html`

paulRbr committed:
Serve kibana on the /kibana route

caruccio commented:
@popox Thanks again for your contribution! Really cool!

paulRbr commented:
@caruccio you're welcome :) !

Love the ELK trio, so making it run on openshift is soo cool.

Glad I could contribute to your cartridge !
